api/src/main/java/org/pmiops/workbench/api/CohortsController.java

```diff
@@ -136,28 +127,61 @@ public void setUserProvider(Provider<User> userProvider) {
       Cohort cohort) {
     // This also enforces registered auth domain.
     workspaceService.enforceWorkspaceAccessLevel(workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
-
     Workspace workspace = workspaceService.getRequired(workspaceNamespace, workspaceId);
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-    org.pmiops.workbench.db.model.Cohort dbCohort = FROM_CLIENT_COHORT.apply(cohort);
-    // TODO: Make this a service method to avoid duplication with WorkspacesController
-    dbCohort.setCreator(userProvider.get());
-    dbCohort.setWorkspaceId(workspace.getWorkspaceId());
-    dbCohort.setCreationTime(now);
-    dbCohort.setLastModifiedTime(now);
-    dbCohort.setVersion(1);
+
+    org.pmiops.workbench.db.model.Cohort newCohort = cohortFactory.createCohort(cohort, userProvider.get(), workspace.getWorkspaceId());
+
+    if (cohortDao.findCohortByNameAndWorkspaceId(newCohort.getName(), workspace.getWorkspaceId()) != null) {
```

Comment:
Can you add this check above the creation via the factory? I think the information should be available there.